### PR TITLE
Fix bsc#1155609 Explain System Level Concept

### PR DESCRIPTION
**Pull request by tomschr**
### Description

This PR fixes bsc#1155609 and contains the following changes:

* In `ha_maintenance.xml`:
  - Explain the recommended way (crm cluster start|stop|restart|status)
  - Refer to SUSE blog for details

* In ha_yast_cluster.xml`:
  - Amend title ("... at the Node Level")
  - Use "cluster service" instead of mentioning Pacemaker (recommended by Yan)
  - Update output of "crm status"

### Checklist

We are on hold to backport this to SLE15SP0/SP1 untill further feedback from customers.

- [ ] To maintenance/SLEHA15
- [ ] To maintenance/SLEHA15SP1

----

@zzhou1 This is a first draft of the above Bugzilla entry. Would you like to review it? I can also create a PDF in case you need it.

### CODE REVIEW

File: xml/ha_maintenance.xml [hunk 1]
```diff
@@ -37,6 +37,50 @@
...
  <sect1 xml:id="sec-ha-maint-shutdown-node">
   <title>Implications of Taking Down a Cluster Node</title>
+  <para>
+   In &sls; &hasi; 15 SP2, the way how cluster services were started and
+   stopped has been changed. &suse; recommends to use the &crmshell;. The
+   old commands with <command>systemctl</command> are still available, but
+   are recommended for experienced users only.
+   For details, refer to the &suse; blog article <link
+    xlink:href="https://www.suse.com/c/suse-high-availability-cluster-services-how-to-stop-start-or-view-the-status/"
+   />.
```
Comment (zzhou1) [inline xml/ha_maintenance.xml:47]: Such link is good reference here to keep HA DOC simple and concise :)

All changes looks good to me

Thanks,
Roger

Comment (tomschr) [reply xml/ha_maintenance.xml:47]: Thanks Roger! :+1:

File: xml/ha_yast_cluster.xml [hunk 2]
```diff
@@ -1027,14 +1027,14 @@ Finished with 1 errors.</screen>
   </sect1>
 
   <sect1 xml:id="sec-ha-installation-start">
-   <title>Bringing the Cluster Online</title>
+   <title>Bringing the Cluster Online at the Node Level</title>
```
Comment (taroth21) [inline xml/ha_yast_cluster.xml:1030]: Initially, I did not get what 'at node level' is supposed to mean, I first needed to look into the bug and the plans for the future as outlined there by Roger. How about 'Bringing the Cluster Online Node by Node'? We can also check with Liam (whom I suggest for additional review anyway) which title would be more clear.

File: xml/ha_yast_cluster.xml [hunk 3]
```diff
@@ -1062,13 +1062,19 @@ Finished with 1 errors.</screen>
...
+  * Last updated: ...
+  * Last change:  ... by hacluster via crmd on &node2;
+  * 2 nodes configured
+  * 1 resource instance configured
 
-Online: [ alice bob ]</screen>
+Node List:
+  * Online: [ &node1; &node2; ]
+
+Full List of Resources:
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/ha_yast_cluster.xml:1076]: Right, I will add an ellipsis to emphasize that something is missing.

File: xml/ha_maintenance.xml [hunk 4]
```diff
@@ -37,6 +37,50 @@
  </info>
  <sect1 xml:id="sec-ha-maint-shutdown-node">
   <title>Implications of Taking Down a Cluster Node</title>
+  <para>
+   In &sls; &hasi; 15 SP2, the way how cluster services were started and
+   stopped has been changed. &suse; recommends to use the &crmshell;. The
```
Comment (lproven) [inline xml/ha_maintenance.xml:42]:
```suggestion
   stopped has been changed. &suse; recommends using the &crmshell;. The
```

File: xml/ha_maintenance.xml [hunk 5]
```diff
@@ -37,6 +37,50 @@
...
+     <screen>crm cluster restart</screen>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Viewing the status of the cluster stack on one node</term>
+    <listitem>
+     <screen>crm cluster status</screen>
+    </listitem>
+   </varlistentry>
+  </variablelist>
```
Comment (lproven) [inline xml/ha_maintenance.xml:78]: I suggest switching around the commands and the explanations of the commands.

So the [term] would be the command, in [screen] tags, and the [listitem] would be the explanation of what it does.

Additionally I recommend removing the definite articles in the explanations:
• stopping cluster services
• restarting cluster services
etc. This is only a _suggestion_ though!

File: xml/ha_yast_cluster.xml (hunk 2 above)
Comment (lproven) [reply xml/ha_yast_cluster.xml:1030]: I prefer @taroth21's title here.

File: xml/ha_maintenance.xml [hunk 6]
```diff
@@ -77,7 +121,7 @@
 <screen>&prompt.root;crm -w node standby</screen>
     <para>
      That way, services can migrate off the node without being limited by the
-     shutdown timeout of &pace;.
+     shutdown timeout of cluster services (&pace;).
```
Comment (lproven) [inline xml/ha_maintenance.xml:124]:
```suggestion
     shutdown timeout of &pace; cluster services.
```

Comment (lproven) [reply xml/ha_maintenance.xml:124]: This is my compromise suggestion between old & new.

I think the word pacemaker, dangling on its own in parentheses on the end of the line, looks odd and does not contribute to the explanation.

File: xml/ha_maintenance.xml (hunk 5 above)
Comment (tomschr) [reply xml/ha_maintenance.xml:78]: Good idea! I'll switch it as adivsed. :+1:

File: xml/ha_yast_cluster.xml (hunk 2 above)
Comment (tomschr) [reply xml/ha_yast_cluster.xml:1030]: Ok, added Tanja's suggestion. Thanks! :+1:

**Comment by tomschr** [comment]
@lproven Thanks Liam for the feedback! Much appreciated! :+1: I've integrated all your suggestions.

File: xml/ha_maintenance.xml [hunk 7]
```diff
@@ -37,6 +37,50 @@
  </info>
  <sect1 xml:id="sec-ha-maint-shutdown-node">
   <title>Implications of Taking Down a Cluster Node</title>
+  <para>
+   In &sls; &hasi; 15 SP2, the way how cluster services were started and
```
Comment (lproven) [inline xml/ha_maintenance.xml:41]:
```suggestion
   In &sls; &hasi; 15 SP2, the way cluster services were started and
```

File: xml/ha_maintenance.xml [hunk 8]
```diff
@@ -37,6 +37,50 @@
...
+   stopped has been changed. &suse; recommends using the &crmshell;. The
```
Comment (lproven) [inline xml/ha_maintenance.xml:42]:
```suggestion
   stopped has changed. &suse; recommends using the &crmshell;. The
```

File: xml/ha_maintenance.xml [hunk 9]
```diff
@@ -37,6 +37,50 @@
...
+   old commands with <command>systemctl</command> are still available, but
+   are recommended for experienced users only.
+   For details, refer to the &suse; blog article <link
+    xlink:href="https://www.suse.com/c/suse-high-availability-cluster-services-how-to-stop-start-or-view-the-status/"
+   />.
+  </para>
+  <para>
+   The recommended way of starting, stopping, or viewing the status is:
```
Comment (lproven) [inline xml/ha_maintenance.xml:50]:
```suggestion
   The recommended way to start, stop, or view the status is:
```

File: xml/ha_maintenance.xml [hunk 10]
```diff
@@ -37,6 +37,50 @@
...
+  </para>
+
+  <variablelist>
+   <varlistentry>
+    <term><command>crm cluster start</command></term>
+    <listitem>
+     <para>Starting cluster services on one node</para>
```
Comment (lproven) [inline xml/ha_maintenance.xml:57]:
```suggestion
     <para>Start cluster services on one node</para>
```

File: xml/ha_maintenance.xml [hunk 11]
```diff
@@ -37,6 +37,50 @@
...
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>crm cluster stop</command></term>
+    <listitem>
+     <para>Stopping cluster services on one node</para>
```
Comment (lproven) [inline xml/ha_maintenance.xml:63]:
```suggestion
     <para>Stop cluster services on one node</para>
```

File: xml/ha_maintenance.xml [hunk 12]
```diff
@@ -37,6 +37,50 @@
...
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>crm cluster restart</command></term>
+    <listitem>
+     <para>Restarting cluster services on one node</para>
```
Comment (lproven) [inline xml/ha_maintenance.xml:69]:
```suggestion
     <para>Restart cluster services on one node</para>
```

File: xml/ha_maintenance.xml [hunk 13]
```diff
@@ -37,6 +37,50 @@
...
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>crm cluster status</command></term>
+    <listitem>
+     <para>Viewing status of the cluster stack on one node</para>
```
Comment (lproven) [inline xml/ha_maintenance.xml:75]:
```suggestion
     <para>View status of the cluster stack on one node</para>
```

File: xml/ha_maintenance.xml [hunk 14]
```diff
@@ -94,7 +138,7 @@ Node <replaceable>&node2;</replaceable>: standby
    </step>
    <step>
     <para>
-     Stop the &pace; service on that node:
+     Stop the cluster services on that node:
```
Comment (lproven) [inline xml/ha_maintenance.xml:141]:
```suggestion
     Stop the &pace; cluster services on that node:
```

File: xml/ha_maintenance.xml [hunk 15]
```diff
@@ -117,7 +161,7 @@ Node <replaceable>&node2;</replaceable>: standby
    </step>
    <step>
     <para>
-     Check if the &pace; service has started:
+     Check if the cluster services has started:
```
Comment (lproven) [inline xml/ha_maintenance.xml:164]:
```suggestion
     Check if the &pace; cluster service has started:
```

Comment (lproven) [reply xml/ha_maintenance.xml:164]: Is the singular OK? Not sure...

File: xml/ha_maintenance.xml (hunk 14 above)
Comment (tomschr) [reply xml/ha_maintenance.xml:141]: I wouldn't say "Pacemaker cluster services" as it's not related entirely on Pacemaker only. There is, for example, Corosync. So Yan Gao recommends to use only "cluster services". I think, that is sufficient. :wink:

File: xml/ha_maintenance.xml (hunk 15 above)
Comment (tomschr) [reply xml/ha_maintenance.xml:164]: That was on purpose. See my comment in https://github.com/SUSE/doc-sleha/pull/132/files#r415655616. It's plural and I would avoid using Pacemaker here. So "cluster services" should be ok

Comment (lproven) [reply xml/ha_maintenance.xml:164]: In that case:

"... Check if the cluster services **have** started:"

**Pull request merged**